src/world.rs

```diff
@@ -225,19 +265,30 @@ impl World {
     }
 
     /// Adds a cube and then recomputes the visibility of the affected cubes (neighbors)
-    fn add_cube(&mut self, at: Vector3, block: Block) {
+    /// Return the cube that need to be rendered 
+    /// and the position of the rendered cube that need to be destroyed
+    fn add_cube(&mut self, at: Vector3, block: Block) -> (Vec<Vector3>,Cube) {
         // For all the neighbors positions, increase their internal counter
         let mut count = 0;
+        let mut to_destroy = Vec::new();
         for pos in Cube::neighbors_positions(at) {
             // Toggle this position
             if let Some(cube_to_toggle) = self.cube_at_mut(pos) {
+
+                // When adding a cube, its neighbors might become non visible
+                // A cube become non visible if it have 5 neighbors before adding a cube
+                // If we test for non visible cube after adding the cube we have extra cube
+                // that were already non visible before
+                if cube_to_toggle.five_neighbors() {
+                    to_destroy.push(cube_to_toggle.position().clone())
+                }
```

Comment:
Let's talk about this. It is not necessary to add a function `has_five_neighbors` to check if a cube has a 5 neighbors right before incrementing it to 6... It adds a layer of complexity. 

Instead, you can just move this block right after the line below ` cube_to_toggle.add_neighhor();` and then you can use `is_visible` 

Therefore you can remove this function `five_neighbors`. This will be better, because it reduces the knowledge of `World` on if a `Cube` is visible...